### ISSUE TITLE
fully move to bicep

### ISSUE THREAD

**Issue by fmms** [Azure/data-management-zone#249]
Hi,

currently the repository contains the bicep templates AND the arm template.

Indeed the compile step is not part of the build pipeline and thus just the pre-compiled ARM template is executed:
https://github.com/Azure/data-management-zone/blob/571c618dbf0ad636e13606b9a558af88d34e5e9c/.ado/workflows/dataManagementZoneDeployment.yml#L60

Shouldn't 
```bash
az bicep build --file .\main.bicep
```

be added to first compile bicep to ARM and then deploy that ...

Thanks for the great repo!

**Comment by marvinbuss** [comment]
We have this on our backlog. The GitHub Actions actually already support the deployment of bicep templates. This is not the case for the ADO Task, but I reached out to the team to get clarity on this to have a consistent experience across ADO and GH.
I will keep you posted.

**Comment by fmms** [comment]
@marvinbuss my idea was pretty much based on https://blog.johnnyreilly.com/2021/03/23/bicep-meet-azure-pipelines-2 .

Adding a 
```yaml
- bash: az bicep build --file infra/main.json
  displayName: 'Compile Bicep to ARM'
```

which then would remove the redundant file already from the repo.

**Comment by marvinbuss** [comment]
I understand what you meant. We will most likely have to do that, but optimally, this is something that would be supported by the task itself, so that we do not have to compile the main.bicep.
Here is the issue, where this gets tracked: https://github.com/microsoft/azure-pipelines-tasks/issues/15337

**Comment by marvinbuss** [comment]
Team is working on implementing the above. Once that is done, we will switch to deploying bicep files instead of ARM.
